core/gatewayclient/client.go

```diff
@@ -137,10 +144,18 @@ func (c *GatewayClient) Call(ctx context.Context, dbid string, action string, in
 	if err == nil {
 		return res, nil
 	}
-	if !errors.Is(err, rpcClient.ErrUnauthorized) {
-		return nil, err
+
+	var jsonRPCErr *jsonrpc.Error
+	if errors.As(err, &jsonRPCErr) {
+		if jsonRPCErr.Code != jsonrpc.ErrorKGWNotAuthorized {
```

Comment:
Ah.  Yup! `core/gatewayclient` can indeed interpret based on the jsonrpc code.

Reply:
Yeah, this is why I have the question https://github.com/kwilteam/kwil-db/pull/700#discussion_r1594780356.

Reply:
In this package, this makes sense.  For the _consumer_ of core/client or core/gatewayclient, they don't _need_ know json rpc codes.  If they want to introspect to see if it's a  json-rpc error that's fine, but it's preferable for external developers to be able to do a simple errors.Is with one of our predefined types.  At least that's how I feel for now.